model-archiver/model_archiver/arg_parser.py

```diff
@@ -19,101 +18,130 @@ class ArgParser(object):
-
-        parser_export.add_argument('-f', '--force',
-                                   required=False,
-                                   action='store_true',
-                                   help='When the -f or --force flag is specified, an existing .mar file with same\n'
-                                        'name as that provided in --model-name in the path specified by --export-path\n'
-                                        'will overwritten')
-
-        parser_export.add_argument('-v', '--version',
-                                   required=True,
-                                   type=str,
-                                   default=None,
-                                   help='Model\'s version')
-
-        parser_export.add_argument('-r', '--requirements-file',
-                                   required=False,
-                                   type=str,
-                                   default=None,
-                                   help='Path to a requirements.txt containing model specific python dependency\n'
-                                        ' packages.')
+        """Argument parser for torch-model-export"""
+
+        parser_export = argparse.ArgumentParser(
+            prog="torch-model-archiver",
+            description="Torch Model Archiver Tool",
+            formatter_class=argparse.RawTextHelpFormatter,
+        )
+
+        parser_export.add_argument(
+            "--model-name",
+            required=True,
+            type=str,
+            default=None,
+            help="Exported model name. Exported file will be named as\n"
+            "model-name.mar and saved in current working directory if no --export-path is\n"
+            "specified, else it will be saved under the export path",
+        )
+
+        parser_export.add_argument(
+            "--serialized-file",
+            required=False,
+            type=str,
+            default=None,
+            help="Path to .pt or .pth file containing state_dict in case of eager mode\n"
+            "or an executable ScriptModule in case of TorchScript\n"
+            "or a .onnx file in the case of ORT."
+            "or a .ts file in the case of tensorRT",
+        )
+
+        parser_export.add_argument(
+            "--model-file",
+            required=False,
+            type=str,
+            default=None,
+            help="Path to python file containing model architecture.\n"
+            "This parameter is mandatory for eager mode models.\n"
+            "The model architecture file must contain only one\n"
+            "class definition extended from torch.nn.modules.",
+        )
+
+        parser_export.add_argument(
+            "--handler",
+            required=True,
+            dest="handler",
+            type=str,
+            default=None,
+            help="TorchServe's default handler name\n"
+            " or Handler path to handle custom inference logic.",
+        )
+
+        parser_export.add_argument(
```

Comment:
Thanks @msaroufim for the PR, so we are thinking of passing the model_config file optional as part of extra file here?

Reply:
Yup that's exactly right